lib/Importer/Caffe2ModelLoader.cpp

```diff
@@ -602,6 +739,72 @@ void Caffe2ModelLoader::loadWeight(const caffe2::OperatorDef &op) {
     return;
   }
 
+  // Load tensors and quantize the values:
+  if (typeName == "Int8GivenTensorFill" ||
+      typeName == "Int8GivenIntTensorFill") {
+    /*
+     output: "conv1_w"
+     name: ""
+     type: "Int8GivenTensorFill"
+     arg {
+     name: "shape"
+     ints: 96
+     ints: 3
+     ints: 11
+     ints: 11
+     }
+     arg {
+     name: "values"
+     s: "\x7f\x80\x80\x7"
+     }
+     arg {
+     name: "Y_scale"
+     f: 0.00044428
+     }
+     arg {
+     name: "Y_zero_point"
+     i: 127
+     }
+     */
+    auto *T = new Tensor();
+    for (auto &o : op.output()) {
+      if (tensors_.count(o))
+        continue;
+      tensors_[o] = T;
+    }
+
+    auto dim = getShape(dict["shape"]);
+
+    assert(dict.count("Y_zero_point") &&
+           "missing zero point for quantzied type");
+    assert(dict.count("Y_scale") && "missing Y_scale for quantzied type");
+
+    float scale = loadFloat(dict["Y_scale"]);
+    int32_t offset = loadInt(dict["Y_zero_point"]);
+    size_t i = 0;
+    if (typeName == "Int8GivenTensorFill") {
+      // Although in Caffe2 quantized model, the weights is int8 quantized,
+      // the weights is stored in uint8_t format due to that Caffe2 requires the
+      // type of input and weights must be the same. Therefore, we need to
+      // convert it to int8 by subtracting 128.
+      T->reset(ElemKind::Int8QTy, dim, scale, offset - OFFSETSHIFT);
+      auto TH = T->getHandle<int8_t>();
+      std::string str = dict["values"]->s();
+      for (; i < str.size(); i++) {
+        TH.raw(i) = ((uint8_t)(str.c_str()[i]) - OFFSETSHIFT);
+      }
+    } else {
+      T->reset(ElemKind::Int32QTy, dim, scale, offset);
```

Comment:
what is the `typeName` for which we set tensor type to be Int32?

Reply:
Yes.

Reply:
`Int8GivenIntTensorFill` this feels a bit strange as Int range is bigger than Int8 range. But since it's what's on C2 we probably cannot do anything.